source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRTextBlockRenderer.mm

```diff
@@ -78,25 +84,49 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
         config->containerStyles.defaultPalette.foregroundColors;
         // Add paragraph style, text color, text weight as attributes to a NSMutableAttributedString, content.
         [content addAttributes:@{NSParagraphStyleAttributeName:paragraphStyle, NSForegroundColorAttributeName:[ACOHostConfig getTextBlockColor:txtBlck->GetTextColor() colorsConfig:colorConfig subtleOption:txtBlck->GetIsSubtle()],} range:NSMakeRange(0, content.length)];
+
+        lab.textContainer.lineBreakMode = NSLineBreakByTruncatingTail;
         lab.attributedText = content;
     }
 
-    lab.numberOfLines = int(txtBlck->GetMaxLines());
-    if(!lab.numberOfLines and !txtBlck->GetWrap()){
-        lab.numberOfLines = 1;
+    lab.area = lab.frame.size.width * lab.frame.size.height;
+
+    ACRContentHoldingUIView *wrappingview = [[ACRContentHoldingUIView alloc] initWithFrame:lab.frame];
+    wrappingview.translatesAutoresizingMaskIntoConstraints = NO;
+    lab.translatesAutoresizingMaskIntoConstraints = NO;
+
+    [viewGroup addArrangedSubview:wrappingview];
+    [wrappingview addSubview:lab];
+
+    NSLayoutAttribute horizontalAlignment = NSLayoutAttributeLeading;
+    if(txtBlck->GetHorizontalAlignment() == HorizontalAlignment::Right) {
+        horizontalAlignment = NSLayoutAttributeTrailing;
+    } else if (txtBlck->GetHorizontalAlignment() == HorizontalAlignment::Center) {
+        horizontalAlignment = NSLayoutAttributeCenterX;
+    }
+
+    [NSLayoutConstraint constraintWithItem:lab attribute:horizontalAlignment relatedBy:NSLayoutRelationEqual toItem:wrappingview attribute:horizontalAlignment multiplier:1.0 constant:0].active = YES;
+    [NSLayoutConstraint constraintWithItem:lab attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:wrappingview attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0].active = YES;
+    [NSLayoutConstraint constraintWithItem:lab attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:wrappingview attribute:NSLayoutAttributeTop multiplier:1.0 constant:0].active = YES;
+
+    lab.textContainer.maximumNumberOfLines = int(txtBlck->GetMaxLines());
+    if(!lab.textContainer.maximumNumberOfLines and !txtBlck->GetWrap()){
```

Comment:
>and [](start = 47, length = 3)

Nit: really prefer to use `&&` operator. Spelling it `and` is legal but so uncommon.